.buildkite/pull-request-pipeline.yml

```
@@ -1,77 +1,28 @@
-# define a GCP VM agent to support container management (by default agent doesn't support)
-agents:
-  provider: "gcp"
-  machineType: "n1-standard-4"
-  image: family/core-ubuntu-2204
-
 steps:
-  # ------------- Unit tests ---------------------
-  - label: ":hammer: Unit tests with LS & ES 8.x-SNAPSHOT :docker:"
-    # Builds the plugin (with current changes) against LS 8.x-SNAPSHOT and ES version defined in gradle.properties
-    # Runs unit tests on LS & ES 8.x-SNAPSHOT docker
-    command:
-      - .buildkite/scripts/run_tests.sh
-    env:
-      ELASTIC_STACK_VERSION: "8.current"
-      SNAPSHOT: true
-      INTEGRATION: false
-
-  - label: ":hammer: Unit tests with LS 8.x-SNAPSHOT & ES main :docker:"
-    # Builds the plugin (with current changes) against LS 8.x-SNAPSHOT and ES main
-    # Runs unit tests on LS & ES 8.x-SNAPSHOT docker
-    command:
-      - .buildkite/scripts/run_tests.sh
-    env:
-      ELASTIC_STACK_VERSION: "8.current"
-      ELASTICSEARCH_TREEISH: main
-      INTEGRATION: false
-      SNAPSHOT: true
-
-  # ------------- Integration tests ---------------------
-  - label: ":hammer: Integration tests with LS & ES 8.x :docker:"
-    # Builds the plugin (with current changes) against LS 8.x and ES version defined in gradle.properties
-    # Runs integration tests on LS & ES 8.x docker
-    command:
-      - .buildkite/scripts/run_tests.sh
-    env:
-      ELASTIC_STACK_VERSION: "8.current"
-      INTEGRATION: true
-      SECURE_INTEGRATION: true
+  - label: "Pull request pipeline"
+    command: |
+      #!/usr/bin/env bash
+      set -eo pipefail
 
-  - label: ":hammer: Integration tests with LS & ES 8.x-SNAPSHOT :docker:"
-    # Builds the plugin (with current changes) against LS 8.x-SNAPSHOT and ES version defined in gradle.properties
-    # Runs integration tests on LS & ES 8.x-SNAPSHOT docker
-    command:
-      - .buildkite/scripts/run_tests.sh
-    env:
-      ELASTIC_STACK_VERSION: "8.current"
-      SNAPSHOT: true
-      INTEGRATION: true
-      SECURE_INTEGRATION: true
-      LOG_LEVEL: "info"
+      echo "--- Downloading prerequisites"
+      python3 -m pip install ruamel.yaml
+      curl -fsSL --retry-max-time 60 --retry 3 --retry-delay 5 -o /usr/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
+      chmod a+x /usr/bin/yq
 
-  - label: ":hammer: Integration tests with LS 8.x-SNAPSHOT & ES main :docker:"
-    # Builds the plugin (with current changes) against LS 8.x-SNAPSHOT and ES main
-    # Runs integration tests on LS & ES 8.x-SNAPSHOT docker
-    command:
-      - .buildkite/scripts/run_tests.sh
-    env:
-      ELASTIC_STACK_VERSION: "8.current"
-      ELASTICSEARCH_TREEISH: "main"
-      SNAPSHOT: true
-      INTEGRATION: true
-      SECURE_INTEGRATION: true
-      LOG_LEVEL: "info"
+      echo "--- Generating dynamic steps"
+      set +e
+      python3 .buildkite/scripts/pull-request-pipeline/generate-steps.py >pipeline_steps.yml
```

nit: readability? is this a  redirect to a file? Space separating may make this more clear that its not an argument to the script (maybe it is though, not sure). 
```suggestion
      python3 .buildkite/scripts/pull-request-pipeline/generate-steps.py > pipeline_steps.yml
```